File: tests/07_phases_ompss_2.dg/c/success_onready_03.c
/*
<testinfo>
test_generator="config/mercurium-ompss-2"
</testinfo>
*/

void bar(int *n) {
    *n = 77;
}

int main() {
    int n = 4;
    int i;
    #pragma oss task for onready(bar(&n)) shared(n)
    for (i = 0; i < 10; ++i)
    {}
    #pragma oss taskwait
    if (n != 77)
        return 1;
    return 0;
}

File: tests/07_phases_ompss_2.dg/fortran/success_onready_03.f90
! <testinfo>
! test_generator=config/mercurium-ompss-2
! test_ignore=yes
! test_ignore_reason="feature not supported by nanos6"
! </testinfo>

SUBROUTINE BAR(X)
IMPLICIT NONE
INTEGER, INTENT(INOUT) :: X
X = 77
END SUBROUTINE

PROGRAM P
IMPLICIT NONE
INTEGER :: X
INTEGER :: I

X = 5

!$OSS TASK DO ONREADY(BAR(X)) SHARED(X)
DO I = 1, 10
CONTINUE
END DO
!$OSS END TASK DO

!$OSS TASKWAIT

IF (X /= 77) STOP 1

END PROGRAM
